lib/puppet/util/windows/adsi.rb

```diff
@@ -312,7 +376,7 @@ def self.uri(name, host = '.')
     end
 
     def native_group
-      @native_group ||= Puppet::Util::Windows::ADSI.connect(uri)
+      @native_group ||= Puppet::Util::Windows::ADSI.connect(self.class.uri(*self.class.parse_name(name)))
```

Comment:
Brings the same behavior for user domain\account checking that is present in users.